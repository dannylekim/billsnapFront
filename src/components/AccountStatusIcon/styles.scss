@import "../../variables";

.user_icon {
  justify-content: center;
}

.status_icon {
  position: absolute;
  right: -12px;
  bottom: 0;
  z-index: 1;
}

.accounts__icons {
  display: flex;
  position: relative;
  margin-right: 1em;
}

.icon__name {
  font-size: 15px;
  text-align: left;
  margin-left: 0.4em;
  font-weight: bold;
  font-family: $primary-font;
  color: $primary-text-color;
  font-style: normal;
  margin-top: -1em;
}
