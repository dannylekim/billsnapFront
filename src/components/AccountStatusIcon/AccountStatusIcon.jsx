import {MdAccountCircle, MdClear, MdDone, MdHourglassFull,} from "react-icons/md/index";
import React from "react";
import "./styles.scss";

/**
 *
 * Takes percent
 *
 * @param status
 * @param sizeMultiplier
 * @returns {JSX.Element}
 */
const getStatusIcons = (status, sizeMultiplier) => {
  switch (status) {
    case "ACCEPTED":
      return (
        <MdDone
          className="status_icon"
          size={40 * sizeMultiplier}
          color="#47E5B6"
        />
      );
    case "DECLINED":
      return (
        <MdClear
          className="status_icon"
          size={30 * sizeMultiplier}
          color="red"
        />
      );
    case "PENDING":
      return (
        <MdHourglassFull
          className="status_icon"
          size={30 * sizeMultiplier}
          color="#FFB400"
        />
      );
    default:
      return <span />;
  }
};

const AccountStatusIcon = ({ status, name, sizeMultiplier = 1 }) => {
  return (
    <div>
      <div className="accounts__icons">
        <MdAccountCircle size={50 * sizeMultiplier} />
        {getStatusIcons(status, sizeMultiplier)}
      </div>
      <p className="icon__name">{name}</p>
    </div>
  );
};

export default AccountStatusIcon;
