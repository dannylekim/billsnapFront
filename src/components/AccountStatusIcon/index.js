import AccountStatusIcon from "./AccountStatusIcon";

export default AccountStatusIcon;