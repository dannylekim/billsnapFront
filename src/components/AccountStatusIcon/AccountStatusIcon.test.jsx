import React from "react";
import AccountStatusIcon from "./AccountStatusIcon";

describe("AccountStatusIcon", () => {
  describe("render", () => {
    describe("snapshots ðŸ“¸", () => {
      it("AccountStatusIcon must match snapshot for ACCEPTED", () => {
        matches(
          <AccountStatusIcon sizeMultiplier={1} status="ACCEPTED" name="TEST" />
        );
      });
      it("AccountStatusIcon must match snapshot for PENDING", () => {
        matches(
          <AccountStatusIcon sizeMultiplier={2} status="PENDING" name="TEST" />
        );
      });
      it("AccountStatusIcon must match snapshot for DECLINED", () => {
        matches(
          <AccountStatusIcon sizeMultiplier={3} status="DECLINED" name="TEST" />
        );
      });

      it("AccountStatusIcon must match snapshot for invalid status", () => {
        matches(
          <AccountStatusIcon sizeMultiplier={3} status="INVALID" name="TEST" />
        );
      });
    });
  });
});
