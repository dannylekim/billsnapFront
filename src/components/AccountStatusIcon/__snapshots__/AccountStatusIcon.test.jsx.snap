// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountStatusIcon render snapshots ðŸ“¸ AccountStatusIcon must match snapshot for ACCEPTED 1`] = `
<div>
  <div
    className="accounts__icons"
  >
    <svg
      fill="currentColor"
      height={50}
      size={50}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={50}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
      />
    </svg>
    <svg
      className="status_icon"
      color="#47E5B6"
      fill="currentColor"
      height={40}
      size={40}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "#47E5B6",
        }
      }
      viewBox="0 0 24 24"
      width={40}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
      />
    </svg>
  </div>
  <p
    className="icon__name"
  >
    TEST
  </p>
</div>
`;

exports[`AccountStatusIcon render snapshots ðŸ“¸ AccountStatusIcon must match snapshot for DECLINED 1`] = `
<div>
  <div
    className="accounts__icons"
  >
    <svg
      fill="currentColor"
      height={150}
      size={150}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={150}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
      />
    </svg>
    <svg
      className="status_icon"
      color="red"
      fill="currentColor"
      height={90}
      size={90}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "red",
        }
      }
      viewBox="0 0 24 24"
      width={90}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
  </div>
  <p
    className="icon__name"
  >
    TEST
  </p>
</div>
`;

exports[`AccountStatusIcon render snapshots ðŸ“¸ AccountStatusIcon must match snapshot for PENDING 1`] = `
<div>
  <div
    className="accounts__icons"
  >
    <svg
      fill="currentColor"
      height={100}
      size={100}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={100}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
      />
    </svg>
    <svg
      className="status_icon"
      color="#FFB400"
      fill="currentColor"
      height={60}
      size={60}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": "#FFB400",
        }
      }
      viewBox="0 0 24 24"
      width={60}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M6 2v6h.01L6 8.01 10 12l-4 4 .01.01H6V22h12v-5.99h-.01L18 16l-4-4 4-3.99-.01-.01H18V2H6z"
      />
    </svg>
  </div>
  <p
    className="icon__name"
  >
    TEST
  </p>
</div>
`;

exports[`AccountStatusIcon render snapshots ðŸ“¸ AccountStatusIcon must match snapshot for invalid status 1`] = `
<div>
  <div
    className="accounts__icons"
  >
    <svg
      fill="currentColor"
      height={150}
      size={150}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={150}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
      />
    </svg>
    <span />
  </div>
  <p
    className="icon__name"
  >
    TEST
  </p>
</div>
`;
